apiserver/facades/controller/charmrevisionupdater/charmhub_test.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrevisionupdater_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/juju/charm/v8"
+	"github.com/juju/errors"
+	"github.com/juju/juju/testcharms"
+	"github.com/juju/names/v4"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facades/controller/charmrevisionupdater"
+	"github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/charmhub"
+	"github.com/juju/juju/charmhub/transport"
+	jujutesting "github.com/juju/juju/juju/testing"
+	"github.com/juju/juju/state"
+)
+
+type charmhubSuite struct {
+	jujutesting.JujuConnSuite
```

Comment:
Per discussion with Ian at our standup, I intend to do this in a follow-up PR as we want to get this PR integrated today, and it would require a fair bit of refactoring and additional work to avoid JujuConnSuite here. Coming soon!

Reply:
We pretty much never want to add new suites that use `JujuConnSuite`.